scripts/scons_helpers/build-3rdparty.py

```diff
@@ -664,18 +664,152 @@ def splitenv(envlist):
+    >>> openssl_get_platform('i686-pc-linux-gnu')
+    'linux-x86'
+
+    # other edge cases
+    >>> openssl_get_platform('mingw-w64-x86_64')
+    'mingw64'
+    >>> openssl_get_platform('x86_64-pc-cygwin')
+    'Cygwin-x86_64'
+    >>> openssl_get_platform('')
+    'cc'
+    """
+
+    # we assume that:
+    # - 'android' has a higher priority than 'linux'
+    # - if toolchain contains '64' and 'android' => we have android64 on 64-bit CPU
+    # - if toolchain contains '64' and 'linux' => we have linux64 on 64-bit CPU
+    # - openssl architectures 'aout', 'elf', 'latomic' are meaningless (they cannot be found in
+    #   toolchain)
+    # - compiler suffixes '-cc', '-gcc', '-clang' are meaningless too
+    # - aix* OSes are never used :)
+    # - 'android64' are currently just aliases for 'android' architectures;
+    #   'android64-aarch64' is the same as 'android-arm64'
+
+    bitness = 64 if '64' in toolchain else 32
+
+    ### generic detection
+
+    # XXX: this is a dirty parser based on very simple heuristics and verified on a
+    # limited set of platforms
+
+    # Set of platforms formatted as (OS, [arch1, ...]).
+    #
+    # `arch` is an architecture that openssl accepts for a given `os`. arch in a toolchain may
+    # be the same but called differently, e.g. 'x86' vs 'i686'. To handle such cases,
+    # `arch_aliases` are used (see below): if `arch` is not in toolchain, we try to search all
+    # aliases (one by one) before moving to the next arch.
+    #
+    # Some tunings:
+    # - 'android-armeabi' is an alias of 'android-arm' => removed 'armeabi'
+    # - 'linux-armv4' fails on android toolchain; we use 'linux-generic32' => removed 'armv4'
+    # - 'android64' is an alias for 'android' => removed 'android64'
+    # - 'generic64' and 'generic32' never occur in toolchains => removed them
+    platforms = [
+        # order is significant here as we search substrings in a whole toolchain string!
+        # [(os,      [arch, ...]), ...]
+        ('BSD',      ['aarch64', 'ia64', 'riscv64',
+                      'sparc64', 'sparcv8', 'x86_64', 'x86']),
+        ('Cygwin',   ['i386', 'i486', 'i586', 'i686', 'x86_64', 'x86']),
+        ('android',  ['arm64', 'arm', 'mips64', 'mips', 'x86_64', 'x86']),
+        ('darwin64', ['arm64', 'ppc', 'x86_64']),
+        ('darwin',   ['i386', 'ppc']),
+        ('linux64',  ['loongarch64', 'mips64', 'riscv64', 's390x', 'sparcv9']),
+        ('linux32',  ['s390x']),
+        ('linux',    ['aarch64', 'arm64ilp32', 'ia64', 'mips64', 'mips32', 'ppc64le',
+                      'ppc64', 'ppc', 'sparcv8', 'sparcv9', 'x86_64', 'x86', 'x32']),
+        ('mingw64',  []),
+        ('mingw',    []),
+    ]
+    arch_aliases = {
+        'arm':       ['armv4', 'armv7a'], # just in case
+        'arm64':     ['aarch64'], # fix for absent platform dirs in android NDK
+        'generic32': ['generic'],
+        'mips32':    ['mips'],
+        'x86':       ['i386', 'i486', 'i586', 'i686'],
+    }
```

Comment:
Let's remove `armv4` and `armv7a` - `arm` is subset of both of them, so should work without alias.

Let's also remove `generic` - it never appears in toolchain.

Let's also remove comment `# fix for absent platform dirs in android NDK` - android is not the only place where aarch64 is used instead of arm64, this name is widely used.

I checked, after removing all them tests still pass.